File: cors/src/main/java/com/cors/GreetingResource.java
package com.cors;

import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class GreetingResource {

    @POST
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello";
    }
}

File: getting-started-kotlin/src/main/kotlin/tech/donau/course/GreetingResource.kt
package tech.donau.course

import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/hello")
class GreetingResource {

    @Inject
    @field: Default
    lateinit var greetingService: GreetingService

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "hello"

    @GET
    @Path("/greetings/{name}")
    fun greet(@PathParam("name") name: String): String{
        return greetingService.greet(name)
    }
}